plugins/module_utils/network/sonic/config/dhcp_snooping/dhcp_snooping.py

```diff
@@ -0,0 +1,641 @@
+
+        result['before'] = existing_dhcp_snooping_facts
+        if result['changed']:
+            result['after'] = changed_dhcp_snooping_facts
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_dhcp_snooping_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+        have = existing_dhcp_snooping_facts
+        resp = self.set_state(want, have)
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        afis = {}
+        want = self.remove_none(want)
+
+        # just in case weird arguments passed
+        if want is None:
+            want = {}
+        if have is None:
+            have = {}
+
+        if want.get('afis') is not None:
+            for want_afi in want.get('afis'):
+                if want_afi.get('afi') == self.ipv4_key:
+                    afis['want_ipv4'] = want_afi
+                elif want_afi.get('afi') == self.ipv6_key:
+                    afis['want_ipv6'] = want_afi
+
+        if have.get('afis') is not None:
+            for have_afi in have.get('afis'):
+                if have_afi.get('afi') == self.ipv4_key:
+                    afis['have_ipv4'] = have_afi
+                elif have_afi.get('afi') == self.ipv6_key:
+                    afis['have_ipv6'] = have_afi
+
+        state = self._module.params['state']
+        if state == 'merged':
+            commands, requests = self._state_merged(want, have)
+        elif state == 'deleted':
+            commands, requests = self._state_deleted(want, have, afis)
+        elif state == 'replaced':
+            commands, requests = self._state_replaced(want, have, afis)
+        elif state == 'overridden':
+            commands, requests = self._state_overridden(want, have, afis)
+
+        return commands, requests
+
+    def _state_merged(self, want, have):
+        """ The command generator when state is merged
+
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        want = remove_empties(want)
+        self.validate_config({"config": want})
+
+        commands = get_diff(want, have, test_keys=self.test_keys)
+        requests = self.get_modify_requests(commands)
```

Comment:
Similar to 'replaced' and 'overridden' states in the 'diff' obtained, missing values in 'source_binding' have to be updated.

Reply:
source bindings uses a patch request, which so far has been able to merge the changes in

Reply:
I see the error in this section now